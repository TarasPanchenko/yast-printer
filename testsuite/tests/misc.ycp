/**
 * File:
 *  misc.ycp
 *
 * Module:
 *  Printer configurator
 *
 * Summary:
 *  Miscelaneous tests.
 *
 * Authors:
 *  Petr Blahos <pblahos@suse.cz>
 *
 * $Id$
 *
 * Miscelaneous tests.
 */

// FIXME: printcapTable testing disabled until UI gets clear

{
    import "Pkg";

    global define UI::OpenDialog(term t)``{}
    global define UI::CloseDialog()``{}
    global define term UI::UserInput()``{return `lprng ;}

    include "testsuite.ycp";
    map I_READ = $[
	"target" : $[
	    "size" : -1,
	    "tmpdir" : "/tmp/",
	    "yast2" : nil,
	],
	"sysconfig" : $[
	    "console" : $[
		"CONSOLE_FONT" : "",
		"CONSOLE_SCREENMAP" : "",
		"CONSOLE_UNICODEMAP" : "",
		"CONSOLE_MAGIC" : "",
		"CONSOLE_ENCODING" : "",
	    ],
	    "language" : $[
		"RC_LANG" : "",
		"DEFAULT_LANGUAGE" : "",
	    ],
	],
	"etc" : $[
	    "install_inf" : $[
		"Cmdline" : "",
		"InstMode" : nil,
	    ],
	],
	"probe" : $[
	    "architecture" : "i386",
	    "has_apm" : true,
	    "has_pcmcia" : false,
	    "has_smp" : false,
	    "system" : [],
	    "memory" : [],
	    "cpu" : [],
	    "cdrom" : $[
		"manual" : [],
	    ],
	    "floppy" : $[
		"manual" : [],
	    ],
	],
    ];
    map I_WRITE = $[];
    map I_EXEC = $[];
    TESTSUITE_INIT ([I_READ, I_WRITE, I_EXEC], 0);
    import "Printer";
    include "printer/misc.ycp";
    TESTSUITE_CLEAN ();

    boolean update_field1 = false;


    map manuf =  $[ "name" : "Ghostscript devices",
                  "printers" : [
                      $[ "name" :  "appledmp",
                       "comment" : "...Apple Dot Matrix Printer 120x72 dpi....",
                       "printerlang" : "",
                       "ps2printer" : $[
                           "params" : [ "-sCOLOR=mono\n-sDEVICE=appledmp\n-r120x72" ]
                           ]
                          ],
                      $[ "name" :  "iwhi",
                       "comment" : "...Apple Imagewriter in high-resolution mode 120x144 dpi....",
                       "printerlang" : "",
                       "ps2printer" : $[
                           "params" : [ "-sCOLOR=mono\n-sDEVICE=iwhi\n-r120x144" ]
                           ]
                          ],
                      $[ "name" :  "iwlo",
                       "comment" : "...Apple Imagewriter in low-resolution mode 120x72 dpi....",
                       "printerlang" : "",
                       "ps2printer" : $[
                           "params" : [ "-sCOLOR=mono\n-sDEVICE=iwlo\n-r120x72" ]
                           ]
                          ],
                      $[ "name" :  "iwlq",
                       "comment" : "...Apple Imagewriter LQ in 320x216 dpi mode....",
                       "printerlang" : "",
                       "ps2printer" : $[
                           "params" : [ "-sCOLOR=mono\n-sDEVICE=iwlo\n-r320x216" ]
                           ]
                          ],
                      $[ "name" :  "t4693d2",
                       "comment" : "...Tektronix 4693d color printer, 2 bits per R/G/B component....",
                       "printerlang" : "",
                       "ps2printer" : $[
                           "params" : [ "-sCOLOR=color\n-sDEVICE=t4693d2\n-r100x100" ]
                           ]
                          ],
                      $[ "name" :  "lq850",
                       "comment" : "...Epson LQ850 printer at 360x360 dpi....",
                       "printerlang" : "",
                       "ps2printer" : $[
                           "params" : [ "-sCOLOR=mono\n-sDEVICE=lq850\n-r360x360" ],
                           "options" : [
                               $[ "macro" : "lprsetupPaperSizes" ]
                               ]
                           ]
                          ],
                      $[ "name" :  "st800",
                       "comment" : "...Epson Stylus 800 printer....",
                       "printerlang" : "",
                       "ps2printer" : $[
                           "params" : [ "-sDEVICE=st800" ],
                           "options" : [
                               $[ "macro" : "lprsetupPaperSizes" ],
                               $[ "macro" : "resolutions360and720" ],
                               $[ "macro" : "colormono" ]
                               ]
                           ]
                          ],
                      $[ "name" :  "cdeskjet",
                       "comment" : "...HP DeskJet 500C (1 bit per pixel color)....",
                       "printerlang" : "",
                       "ps2printer" : $[
                           "params" : [ "-sCOLOR=color\n-sDEVICE=cdeskjet" ],
                           "options" : [
                               $[ "macro" : "lprsetupPaperSizes" ],
                               $[ "macro" : "resolutions300and600" ]
                               ]
                           ]
                          ],
                      $[ "name" :  "cdjcolor",
                       "comment" : "...HP DeskJet 500C (24 bits per pixel color , Floyd-Steinberg dithering)....",
                       "printerlang" : "",
                       "ps2printer" : $[
                           "params" : [ "-sCOLOR=color\n-sDEVICE=cdjcolor" ],
                           "options" : [
                               $[ "macro" : "lprsetupPaperSizes" ],
                               $[ "macro" : "resolutions300and600" ]
                               ]
                           ]
                          ],
                      $[ "name" :  "cdjmono",
                       "comment" : "...HP DeskJets (monochrome)....",
                       "printerlang" : "",
                       "ps2printer" : $[
                           "params" : [ "-sCOLOR=mono\n-sDEVICE=cdjmono" ],
                           "options" : [
                               $[ "macro" : "lprsetupPaperSizes" ],
                               $[ "macro" : "resolutions300and600" ]
                               ]
                           ]
                          ]

                      ]
        ];

    ////////////////////////////////////////////////////////////////
    map user_settings = $[
        "printcap":[
            $[
                "type" : "aps_remote",
                "names": [ "lp", "lp-ascii" ],
                "connection" : $[ "type" : "parallel" ]
                ],
                $[
                    "type" : "aps_yast2",
                    "names": [ "printer1", "printer1-ascii" ],
                    "connection" : $[ "type" : "usb" ]
                    ],
                $[
                    "type" : "other",
                    "names": [ "printer2", "printer2-ascii" ],
                    "connection" : $[ "type" : "usb" ]
                    ],
                $[
                    "type" : "aps_yast2",
                    "names": [ "printer3", "printer3-ascii" ],
                    "connection" : $[ "type" : "serial" ]
                    ]
            ]
        ];
    user_settings = add (user_settings, "default", "lp");
    ////////////////////////////////////////////////////////////////

    map connections = $[
        // up to 40 chars
        "parallel": $[  "name" :    "Parallel printer"
            ],
        // up to 40 chars
        "serial": $[    "name" :    "Serial printer"
            ],
        // up to 40 chars
        "usb": $[       "name" :    "USB printer"
            ],
        // up to 40 chars
        "remote": $[    "name" :    "TCP/IP forwarding queue"
            ],
        // up to 40 chars
        "samba": $[     "name" :    "Samba/Windows printer"
            ],
        // up to 40 chars
        "novell": $[    "name" :    "Novell printer"
            ],
        // up to 40 chars
        "prefilter": $[ "name" :    "Prefilter queue for remote printers"
            ]
        ];

    ////////////////////////////////////////////////////////////////
    map READ = $[];
    ////////////////////////////////////////////////////////////////
    boolean test_mode = false;
    ////////////////////////////////////////////////////////////////
    READ = $[
        "target" : $[ "tmpdir" : "/tmp/YaST2-XXXX" ]
        ];
    TEST (``(diffUppFiles ("string1", "string2")), [READ], nil);
    map EXEC = $[
        "target" : $[ "bash" : 1 ]
        ];
    TEST (``(diffUppFiles ("string1", "string2")), [READ,$[],EXEC], nil);
}
