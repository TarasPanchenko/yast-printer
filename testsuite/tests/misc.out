Read	.probe.architecture "i386"
Read	.probe.has_apm true
Read	.probe.has_pcmcia false
Read	.probe.is_uml false
Read	.probe.has_smp false
Read	.probe.system []
Read	.target.size "/etc/install.inf" -1
Read	.target.tmpdir "/tmp/"
Read	.sysconfig.console.CONSOLE_ENCODING ""
Execute	.target.bash_output "locale -k charmap" $[]
Read	.target.tmpdir "/tmp/YaST2-XXXX"
Write	.target.string "/tmp/YaST2-XXXX/options1.upp" "string1" true
Write	.target.string "/tmp/YaST2-XXXX/options2.upp" "string2" true
Execute	.target.bash "/usr/bin/diff -w -B /tmp/YaST2-XXXX/options1.upp /tmp/YaST2-XXXX/options2.upp" 0
Execute	.target.bash "/bin/rm /tmp/YaST2-XXXX/options1.upp /tmp/YaST2-XXXX/options2.upp" 0
Return	true
Read	.target.tmpdir "/tmp/YaST2-XXXX"
Write	.target.string "/tmp/YaST2-XXXX/options1.upp" "string1" true
Write	.target.string "/tmp/YaST2-XXXX/options2.upp" "string2" true
Execute	.target.bash "/usr/bin/diff -w -B /tmp/YaST2-XXXX/options1.upp /tmp/YaST2-XXXX/options2.upp" 1
Execute	.target.bash "/bin/rm /tmp/YaST2-XXXX/options1.upp /tmp/YaST2-XXXX/options2.upp" 1
Return	false
